### DISCUSSION TITLE
Issues with the AIdeck

### DISCUSSION THREAD

**Discussion by matsarto99** [bitcraze/discussions#825] [Q&A]
Hello,

Sorry for the general title, but I am really getting several issues and don't know where they come from.
I recently changed AIdeck because I accidentally broke the wifi antenna of the old one and it is giving me more trouble than ever.

I followed the steps to flash the bootloader with the Olimex adapter, and that worked as I got the same result as the tutorial. I could also flash code on GAP8, for example, the suggested wifi example .bin in the tutorial.
The problems started putting back on the Flow deck. I could not flash my code on GAP8 anymore, always stopping at 6%. After trying many times to reflash the bootloader, update the firmware etc, still I couldn't flash. The only way I can do that is by removing the flow deck.

I thought I had solved the problem, I don't have to flash often, so I don't care about having to remove the deck. I have the wifi streamer example on the GAP8 and a custom app on the CF. I stream images on the PC, run an AI model, and return the results to the drone via radio.
I am making experiments, sometimes they work, and other times I get issues that I believe are related to the AIdeck. Issues being:
- The images are reversed (the image streamer probably didn't manage to flip them). To solve this I either turn the drone on and off multiple times until it works. Usually, it works more when I turn on the drone while it is not on the ground (really can't understand why) or when it is connected to the PC
- The LED2 on the AIdeck stops blinking and the wifi AP stops working. The only way I managed to solve this is by removing the flow deck, and restarting the drone many times, entering the bootloader mode if necessary. It works better when I plug in the USB. Sometimes nothing works and I have to reflash the GAP8.
- I stop receiving images after the drone starts flying

Since the solutions work better when the drone is attached to the PC I suspect that there might be a problem with the battery but I have it fully charged and the battery test is positive.

All the issues happen pretty often, sometimes one after the other and my experiments are getting endless. Did you ever experience anything similar?

Thanks in advance

Mattia

**Comment by ataffanel** [comment]
Hi, sorry that you are having to many problems with the AI deck. It does indeed looks like some kind of power supply issue.

Unfortunately I do not know much about the AI-deck and I am the only one from Bitcraze not in vacation this week. @tobbeanton will be back on week 31, he will likely have an idea of what could be happening.

**Comment by knmcguire** [comment]
Hi! There have been reports of some of these issues but not all at once as you are experiencing... Let me try to give some thought of each of the issues:

- Reversed images: This also might be due to the resetting done with the AIdeck driver on the crazyflie firmware that perhaps this is done at the wrong time. Perhaps play around with the timers for that: https://github.com/bitcraze/crazyflie-firmware/blob/master/src/deck/drivers/src/aideck.c
- LED2 on the aideck. This seems like a power issue to be honest... 
- Same thing here. The battery is probably not good anymore as the motors are drawing so much power at startup that it stops providing power for the aideck. What you could try though, is to have the propellers already spin before takeoff, then the motors shouldn't draw too much. But perhaps try another battery if you have one? 

The fact that the flowdeck is interfering with the aideck is also worrysome. Yes, flashing with 2 decks is a know issue, but it should have been fixed recently: https://github.com/bitcraze/crazyflie-clients-python/issues/616. Do you have a spare flowdeck to try out as well?

**Reply by matsarto99** [reply]
Hi Kimberly,

I don't really know what to change in the aideck.c file, would you mind giving some more pointers?

I tried changing battery, but the issue with LED2 and interference with flowdeck is still there. Unfortunately, I don't have another flowdeck at the moment. Also, for the interference issue, I tried to bootload again with the flowdeck on, but the flashing stops after updating the mcu on the drone. It actually searches for decks to update, finds the AI deck and then it blocks. After that, LED1 on the AI deck stops and if I try to connect to the drone I cannot display anything on the console, it is like it is still in bootloader mode and hanging there.

Connected to the issues I am facing: I am streaming jpeg images from the AI deck and when the drone is on the ground, all works okay, but when it is flying I either stop receiving images (even some seconds after take-off when I expect motors to draw less power) or I start receiving something, then it stops for some hundred milliseconds, and then it restarts again and I receive suddenly more images than expected, like if they were on a buffer waiting to be sent. 
I can see that both displaying the images and looking at the fps. When displaying the images I notice the streaming going at normal fps (around 8, cannot get more in normal conditions), then it stops and then I receive a "faster" sequence of images. Looking at the printed fps on the terminal, I can see that when the streaming restarts, I can get up to 13-14 fps, higher the streaming rate from the drone.
Does this also have to do with power issues?

Thanks,

Mattia

**Reply by knmcguire** [reply]
- Flowdeck and aideck flashing... then I think that is unfortunately the limitation for now. This has been an issue that sometimes comes back so I don't know how to solve that. However, the cfclient only flashes the ESP (nina module), so if it is skipping that for the next times I think that will be okay as it does not need to be flashed the same program over and over again. The only thing is that will be annoying is if it also blocks the gap8 flashing as well (like for [the wifi example](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#flash-wifi-example)). If that does not happen then no worries, then only the updating of the ESP is a minor inconvenience I say.  

The speed of the wifi images or that it suddenly stops completely or just for a bit... that is perhaps a bit more in the WiFi magic than I can help out with. You could perhaps start from small steps with that:

1. Try streaming without flying
2. Try streaming with slowly ramping up the motors. You can use this [cflib example ](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/motors/ramp.py) or through [cfclient's parameters tab](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/parameters_tab/) with [MotorPowerSet](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#motorpowerset )
3. Try flying at different altitudes. taking off to 0.5 meters takes way less power than flying to 1.5 meters.

I do want to add that The aideck is a difficult deck... so I think it is extremely important to take things at small steps and to understand each step what is happening (like what is flashing what, what does CPX do with WiFi). It unfortunately needs quite some embedded programming experience to work properly with it. I hope that the documentation will help with that a bit but just to set your expectations of trying too much at the beginning https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/

In a few weeks more of my colleagues should be back so there will be more ways to help you with this.
